build/dsc/templates/dsc_type.rb.erb

```diff
@@ -108,7 +108,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
           .select { |c| c[:klass].name == property.embeddedinstance_class_name }
           .first[:klass]
           .features
-          .map { |p| Dsc::Property.new(p) }
+          .map { |p| Dsc::Property.new(p, resource.relative_mof_path) }
```

Comment:
Without this being passed through, the Property can only emit error information with only the property name e.g. 'Property 'Ensure' is not valid'.  This is not useful during debugging as it does not tell you which of the 200+ MOF files have the property, particularly in our case which was debugging the Ensure property.  By passing the full path to the schema MOF file, useful error messages can be generated.